kernel/src/common/utils.rs

```diff
@@ -90,3 +90,19 @@ macro_rules! create_capability {
         Cap
     };};
 }
+
+/// Count the number of passed expressions.
+/// Useful for constructing variable sized arrays in other macros.
+/// Taken from the Little Book of Rust Macros
+///
+/// ```ignore
+/// use kernel:count_exprs;
+///
+/// let count: usize = count_exprs!(1+2, 3+4);
+/// ```
+#[macro_export]
+macro_rules! count_exprs {
```

Comment:
I would prefer this to be `count_expressions` but I can understand if it is better to match the Rust shorthand.

Reply:
I don't particularly care either way, count_expressions it is